iothub/device/src/Common/Exceptions/QuotaExceededException.cs

```diff
@@ -7,8 +7,11 @@
 namespace Microsoft.Azure.Devices.Client.Exceptions
 {
     /// <summary>
-    /// The exception that is thrown when an attempt to add a device fails because the maximum number of registered devices has been reached.
+    /// The exception that is thrown by the device client when the daily message quota for the IoT hub is exceeded.
```

Comment:
This is only thrown as an internal exception because we did not want to make breaking changes. https://github.com/Azure/azure-iot-sdk-csharp/blob/master/iothub/device/src/Transport/AmqpIot/AmqpIotErrorAdapter.cs#L240